examples/linear_model/plot_ridge_coeffs.py

```diff
@@ -1,89 +1,272 @@
+penalizes large training weights, forcing the model to shrink certain
+coefficients. Regularization reduces a model's reliance on specific information
+obtained from the training samples.
+
+This example illustrates how L2 regularization in a
+:class:`~sklearn.linear_model.Ridge` regression affects a model's performance by
+adding a penalty term to the Loss that increases with the coefficients (β).
+
+The regularized loss function is given by: :math:`\\mathcal{L}(X, y, \\beta) = \\|
+y - X \\beta \\|^{2}_{2} + \\alpha \\| \\beta \\|^{2}_{2}`
+
+where :math:`X` is the input data, :math:`y` is the target variable,
+:math:`\\beta` is the vector of coefficients associated with the features, and
+:math:`\\alpha` is the regularization strength.
+
+The regularized loss function aims to balance the trade-off between accurately
+predicting the training set and to prevent overfitting.
+
+In this regularized loss, the left-hand side (:math:`\\|y - X\\beta\\|^{2}_{2}`)
+measures the squared difference between the actual target variable, :math:`y`,
+and the predicted values. Minimizing this term alone could lead to overfitting,
+as the model may become too complex and sensitive to noise in the training data.
+
+To address overfitting, Ridge regularization adds a constraint, called a penalty
+term, (:math:`\\alpha \\| \\beta\\|^{2}_{2}`) to the loss function. This penalty
+term is the sum of the squares of the model's coefficients, multiplied by the
+regularization strength :math:`\\alpha`. By introducing this constraint, Ridge
+regularization discourages any single coefficient :math:`\\beta_{i}` from taking
+an excessively large value and encourages smaller and more evenly distributed
+coefficients. Higher values of :math:`\\alpha` force the coefficients towards
+zero. However, an excessively high :math:`\\alpha` can result in an underfit
+model that fails to capture important patterns in the data.
+
+Therefore, the regularized loss function combines the prediction accuracy term
+and the penalty term. By adjusting the regularization strength, practitioners
+can fine-tune the degree of constraint imposed on the weights, training a model
+capable of generalizing well to unseen data while avoiding overfitting.
+
+For a comparison between Ridge and Huber Regularization see
+:doc:`~examples.linear_model.plot_huber_vs_ridge`
 """
 
 # Author: Kornel Kielczewski -- <kornel.k@plusnet.pl>
 
-import matplotlib.pyplot as plt
-import numpy as np
+# %%
+# Purpose of this example
+# -----------------------
+# For the purpose of showing how Ridge regularization works, we will create two
+# data sets: non-noisy and noisy. Then we will train them on a range of
+# regularization strengths (:math:`\alpha`) and plot how the trained
+# coefficients and the mean squared error between those and the original values
+# behave as functions of the regularization strength.
+
+# %%
+# Ridge Regularization on non-noisy data set
+# ------------------------------------------
 
+# %%
```

Comment:
```suggestion
# ------------------------------------------
#
#
```

Reply:
Hm, the problem is that when I do that, the headings' formatting goes wild (at least in this second case of putting ------------- and ######## underlined headings into the same cell) and when building the doc I also get the following warnings and criticals:
```
/home/stefanie/Python/scikit-learn_dev/scikit-learn/doc/auto_examples/linear_model/plot_ridge_coeffs.rst:92: CRITICAL: Unexpected section title or transition.

-----------------------
/home/stefanie/Python/scikit-learn_dev/scikit-learn/doc/auto_examples/linear_model/plot_ridge_coeffs.rst:100: CRITICAL: Unexpected section title.

Ridge Regularization on non-noisy data set
------------------------------------------
/home/stefanie/Python/scikit-learn_dev/scikit-learn/doc/auto_examples/linear_model/plot_ridge_coeffs.rst:104: WARNING: Definition list ends without a blank line; unexpected unindent.
/home/stefanie/Python/scikit-learn_dev/scikit-learn/doc/auto_examples/linear_model/plot_ridge_coeffs.rst:104: CRITICAL: Missing matching underline for section title overline.

```
Whereas if I keep the cells separate everything's fine.
Am I doing something wrong when I unify the cells like this:
```
# %%
# Purpose of this example
# -----------------------
# For the purpose of showing how Ridge regularization works, we will create a
# non-noisy data set. Then we will train a regularized model on a range of
# regularization strengths (:math:`\alpha`) and plot how the trained
# coefficients and the mean squared error between those and the original values
# behave as functions of the regularization strength.
#
# Ridge Regularization on non-noisy data set
# ------------------------------------------
# 
# Creating non-noisy data set
#############################
# We make a toy data set, that's suitable to detect regression. Please note that
# in this example the data is non-noisy, hence it is possible to extract the
# exact coefficients.
```